stix_shifter/stix_transmission/src/modules/qradar/RestApiClient.py

```diff
@@ -1,189 +0,0 @@
-from urllib.request import install_opener
-from urllib.request import build_opener
-from urllib.request import HTTPSHandler
-from .Utilities import *
-
-import ssl
-import sys
-import base64
-import os
-
-# This is a simple HTTP client that can be used to access the REST API
-class RestApiClient:
-
-    # Constructor for the RestApiClient Class
-    def __init__(self, server_ip, auth_token, cert=None, proxy=None, version=None):
-        self.x_forward_proxy = None
-        self.x_forward_proxy_auth = None
-
-        self.headers = {'Accept': 'application/json'}
-        if version is not None:
-            self.headers['Version'] = version
-        if auth_token:
-            self.headers['SEC'] = auth_token
-        else:
-            raise Exception('No valid credentials found in configuration.')
-
-        if proxy is not None:
-            proxy_url = proxy.get('url')
-            proxy_auth = proxy.get('auth')
-
-            if (proxy_url is not None and
-                    proxy_auth is not None):
-                self.headers['proxy'] = proxy_url
-                self.headers['Proxy-Authorization'] = 'Basic ' + proxy_auth
-
-            self.x_forward_proxy = proxy.get('x_forward_proxy')
-            self.x_forward_proxy_auth = proxy.get('x_forward_proxy_auth')
-
-        self.server_ip = server_ip
-        self.base_uri = '/api/'
-
-        # Create a secure SSLContext
-        # PROTOCOL_SSLv23 is misleading.  PROTOCOL_SSLv23 will use the highest
-        # version of SSL or TLS that both the client and server supports.
-        context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
-
-        # SSL version 2 and SSL version 3 are insecure. The insecure versions
-        # are disabled.
-        context.verify_mode = ssl.CERT_REQUIRED
-        if sys.version_info >= (3, 4):
-            context.check_hostname = True
-
-        check_hostname = True
-        if cert is not None:
-            # put cert/key pair into a file to read it later
-            cert_file_name = "cert.pem"
-            certfile = open("cert.pem", "w+")
-            certfile.write(cert)
-            certfile.close()
-
-            with open(cert_file_name, 'w+') as f:
-                try:
-                    f.write(cert)
-                    f.close()
-                except IOError:
-                    print('Failed to setup certificate')
-
-            # The default QRadar certificate does not have a valid hostname,
-            # so me must disable hostname checking.
-            if sys.version_info >= (3, 4):
-                context.check_hostname = False
-            check_hostname = False
-
-            # Instead of loading the default certificates load only the
-            # certificates specified by the user.
-            try:
-                context.load_default_certs()
-                context.load_cert_chain(certfile="cert.pem")
-            except:
-                print('Failed to load certificate')
-
-            os.remove("cert.pem")
```

Comment:
We want to keep this line in the new RestAPIClient, I'll comment in the new file

Reply:
the deletion now is executed in the destructor. (__del__)